flocker/node/functional/test_docker.py

```diff
@@ -367,6 +367,39 @@ def added(_):
         d.addCallback(added)
         return d
 
+    def test_null_environment(self):
+        """
+        A container that does not include any environment variables contains
+        an empty ``environment`` in the return ``Unit``.
+        """
+        docker_dir = FilePath(self.mktemp())
+        docker_dir.makedirs()
+        docker_dir.child(b"Dockerfile").setContent(
+            b'FROM scratch\n'
+            b'MAINTAINER info@clusterhq.com\n'
+            b'CMD ["/bin/doesnotexist"]'
+        )
+        image = DockerImageBuilder(test=self, source_dir=docker_dir)
+        image_name = image.build()
+        client = self.make_client()
+        name = random_name()
+        if isinstance(client, NamespacedDockerClient):
+            container_name = client._client._to_container_name(name)
+            client._client._client.create_container(
+                name=container_name, image=image_name)
+        else:
+            container_name = client._to_container_name(name)
+            client._client.create_container(
+                name=container_name, image=image_name)
```

Comment:
Instead of this `isinstance` check, add a new hook to `GenericDockerClientTests`: `create_container(client, name, image)`.  The first branch of this conditional is the implementation of that method for `NamespacedDockerClientTests`.  The second branch of this conditional is the implementation for `GenericDockerClientTests`.